app/models/nic/base.rb

```diff
@@ -45,6 +45,8 @@ class Base < ApplicationRecord
     validates :subnet, :belongs_to_host_taxonomy => { :taxonomy => :organization }
     validates :subnet6, :belongs_to_host_taxonomy => { :taxonomy => :organization }
 
+    before_create check_blank_mac_for_virtual_resources
```

Comment:
I don't think adding validation error on before_create will work, alsow, you're running the method right away, instead of using the symbol.
Why dont you use following?
```suggestion
    validate :check_blank_mac_for_virtual_resources
```